### ISSUE TITLE
Multiblock Balancing ft. 0lafe

### ISSUE THREAD

**Issue by Bluebine** [draknyte1/GTplusplus#430]
A quick condensation of all the ideas discussed in the GTNH discord:

1. Remove the speed bonuses on the GT++ multiblock, and change the parallelism to 4* tiers above the recipe up to 64x.
So with an LV recipe, it'd do 1x parallel with LV power, 4x parallel with MV power, 16x parallel with HV power, 64x with EV+ power.
Universally agreed on in our conversation.

2. To balance with the previous suggestion, remove control cores altogether, go back to the old system of just upgrading energy hatches.
Agreed about by everyone except for 0lafe.

3. As an alternate to #2, change the recipe for control cores.
Per tier:
4x superconductor of tier
4x tier Power IC (ultra low for ULV-MV, low power for HV, regular for EV, etc)
4x circuits of tier+1
16x tier material, 16x alt tier material (still the same materials as before, just a bit less of them)
EV+ have 2x tier emitter
LuV+ have 2x tier field gen
UV+ have undecided extra cost- maybe something from the cyclotron?
0lafe's suggestion for control cores.

4. Change how control cores work- now control the max tier of recipe that can be done in the multiblocks, instead of max power allowed in.
Alk's idea if I remember correctly, most people in the discord support, 0lafe does not.

EDIT: 5. Tier materials need looked at, with regard to ABS recipes. ZPM cores are uncraftable at the moment, since they require HG-1223, which needs ZPM power to make. Every core HV and below has the same problem- their alloys require you to already HAVE a control core.
This would need to be fixed to make the cores usable.

**Comment by Paolkageist** [comment]
1.  Good.  Nerfing speed doesn't affect whether existing multiblocks can actually be formed or used.  Doesn't break anyone's existing setups like the control core addition does.  
2.  Good.  Fuck control cores.  Hatches cost more now, let's just go with those.  If you want to overclock with multiple power hatches, then that's consistent with vanilla GT multiblocks.  
3.  Fuck control cores.
4.  See 3.

**Comment by Anarack** [comment]
1. I have no opinion on.
2. Control cores look a bit nasty recipe wise, instead of the core you could simply use the energy hatch to set the min tier of bus/hatches required with the downside of having the craft a lot of hatches when upgrading with the max hatch required toping out a UHV/MAX like the rest of your stuff.
3. see 2.
4. This sounds better as with the other the low tier cores are pointless. except for a few machines.

**Comment by kekzdealer** [comment]
I vote for #1 & #2. It's a clear design and it would still be worth to have the multis. I actually like that more than the current seemingly arbitrary speed bonuses.

Would love if they kept the "Only uses 80% of EU" but that's just my biased opinion because I want to use them for power generation where efficiency is important.

**Comment by 0lafe** [comment]
Locking them to hatches other than energy is just annoying. There's times you want a ULV bus over anything, just for the size not the cost. My reasoning for cores is to give the multis a sizable upgrade cost, which the energy hatch doesn't really cover, nor do I wish it to. Otherwise an IV multi is only different than a ZPM multi because of a fairly easy to craft hatch, so mostly the tier of power. Imo it makes progressing a little too easy when the hatch cost is so low.

**Comment by Paolkageist** [comment]
So increase the cost of energy hatches.  Don't keep changing how the multiblocks are formed.  Or do you actually want to add control cores to every multiblock, not just GT++?

**Comment by AlexBucket** [comment]
As someone just entering UV machine era, having used GT++ multis the last few months.

1. Meh. I wouldn't remove the speed increase all together. They're a multi, and I like the parrellism + speed boost. Could be balanced a bit more.

2. Fuck control cores, go back to energy hatch setting tier.

3. Fuck control cores again.

4. Fuck control cores.


I can get where 0lafe and the nerf-brigade are coming from; the multis themselves and quite cheap to make. They start at IV and are basically comparable (at a guess) to a ZPM single block, at least. Make the casings more expensive, or move the machine the multi is based around to LuV or ZPM which then gates it behind the assembly line. Coupled with the energy hatch changes (which I think will be fine once we get used to them) will mean upgrading the power tier isn't as cheap as it is now in 2.,0.5.1

**Comment by 0lafe** [comment]
@pao only the ones with "enhanced overclocking" so like ones that get a bonus with a new hatch. I don't think things like the ebf need it because they don't get anything for upgrading besides opening new recipes. However if a multi has parallels that self expand with tiering them up, I think it should come with a cost of tiering them up. Energy hatches shouldn't be that cost imo because then everything is effected equally, despite some multis benefiting significantly more than others. This mostly partains to things that are upgraded versions of other stuff. Either multis or singles. These are much more viable in higher tiers, and I think upgrading them should reflect that. An IV multi with a new hatch isn't really a higher tier multi, and I don't like the idea of most multis being EV/IV and then staying that way all the way to UHV+. So basically I want them on the ++ versions of singles, the ++ versions of other multis, and if possible the LCR because damn that thing be good. This gives a cost to the increased speed. In terms of singleblocks, upgrading a single replacement multi gives you 1 of that tier, and multiplies your quantity of the tiers before it by 4. I think the cost of this is more than fairly displayed in the core recipe. 

>Make the casings more expensive, or move the machine the multi is based around to LuV or ZPM which then gates it behind the assembly line

Maybe, but I'm trying to give these a persistent cost that sticks with them when tiering up. Something the energy hatch fails to do, because of the wide use it has. If the cost is purely on the power, then tiering up your base only consists of generating more power, which makes things a whole lot easier in the later tiers, which doesn't make sense to me.

**Comment by AlexBucket** [comment]
Why does it need to be so exponentially expensive though? You're just going to turn people away from using the whole mod because it's rediulously expensive for no real reason.
It should be that you've reach this milestone of LuV/ZPM and now these wonderful multis are available to you. I say the cost of the energy hatches will be a good cost to them tiering up, if they're reciped right, along side making the controller itself require LuV/ZPM machines and some more expensive metals for the casings. Therefore you have the expensive outlay to start, and have the cost of upgrading the energy hatch instead of just making a single block and being on your merry way.

**Comment by Paolkageist** [comment]
How about this:  take all the ideas about rebalancing, retiering, etc., plan out how you think the mod should be.  Then save them for GTNH 3.0.  Having to deal with a new set of nerfs each release is ridiculously disruptive, and it'll put off people from updating until they start a new world.  Saving them for 3.0, when people would be starting a new world anyway, means you can implement them all you want, all at the same time, without fucking up everyone's existing setups. I don't want to have to choose between getting new quests and bug fixes, and having to redo every multiblock I've made and recreating every AE2 autocrafting recipe I've set up just because you feel the need to push out these nerfs in the middle of the pack.  Leave the nerfs out of the small modpack updates.  They don't *have* to be done right now.

**Comment by finkledoodle** [comment]
The current system is fine as of 5.1. 90% of players quit this pack before even getting to the point where they can consider GT++ multiblocks, and many that get that far don't bother because nearly every GT++ machine causes pollution while their single-block comparisons don't. I'm not sure why there is so much debate over GT++ multiblocks when things like processing arrays have existed for years now, and I don't see anyone ever commenting about how OP they are. As Alex said, just increase the cost of the control block instead of adding these cores, making sure they are expensive up front. I also agree with Pao, redoing setups that took weeks to figure out and design, and that I have relied on for extended periods of time is extremely disruptive. Even if these ideas were saved for 3.0 though, some of us will still be playing on existing, older worlds because of how long this pack takes. I've been on the same world for over a year now and still haven't finished this pack, and each update only slows down newer players from getting anywhere close. I'd really prefer people focus on creating new content rather than tinkering with existing things that I don't hear anyone else have issues with.

**Comment by 0lafe** [comment]
>I'm not sure why there is so much debate over GT++ multiblocks when things like processing arrays have existed for years now, and I don't see anyone ever commenting about how OP they are

Because PAs follow GT rules. You get out what you put in. They just make inventory management of 64 machines easier, and take up less space. But you still fill them with singles to make up whatever you're doing. The speed it runs at is exactly the speed of the singles inside it. The ++ multis run at up to 6x faster than single blocks, which means doing a tier recipe they can be up to 6x faster, which is something that's *impossible* otherwise. PAs don't add anything you couldn't do before, they simply make the item logistics simpler and don't take 64 blocks. 

>and many that get that far don't bother because nearly every GT++ machine causes pollution while their single-block comparisons don't. 

Pollution is not a big deterrent, and the quest book tries to show you that these are good upgrades and machines you should use. If they choose not to then that's personal preference, but it doesn't have much bearing on the usefulness of the machine. 

> Pao, redoing setups that took weeks to figure out and design, and that I have relied on for extended periods of time is extremely disruptive. 

The only things I can see for this would be ore processing or fusion, both of which won't change much as these run on parallel recipes. You may need to add some more multis, or maybe less in some small cases, but the change here is very minimal. For the most part you wouldn't even need to change anything, you'd just get different speeds after this change. 

>some of us will still be playing on existing, older worlds because of how long this pack takes.

Not for 3.0, that's supposed to be 1.12

>I'd really prefer people focus on creating new content rather than tinkering with existing things that I don't hear anyone else have issues with

Theres still an extensive list of new features/machines and stuff in 2.0.6.8 from 2.0.6.2, and still some in that you don't see in 2.0.5.1. Changing mechanics isn't about limiting additions, it's about making those additions fit to NH. We don't just add new mods and not change them, that's the whole point of this pack, everything is changed to create a balanced progression system. Something almost every other modpack fails to do in the same way. 

>Having to deal with a new set of nerfs each release is ridiculously disruptive, and it'll put off people from updating until they start a new world. 

If they want to play like that then they can, but the intended way to play is to update and deal with the changes, you're not really meant to have a choice about it. That's why the official server, and all the unofficials iirc, will update to the newest stable version. If you really want to wait until you restart to deal with changing recipes then you're free to do so, but the game shouldnt be made around those people. 

>and having to redo every multiblock I've made and recreating every AE2 autocrafting recipe I've set up just because you feel the need to push out these nerfs in the middle of the pack

You can't nerf at the start then add new things and expect no nerfs. The whole point is to keep things even, and when you add new mechanics without changing them, you often end up with these issues. It would have been ideal to change the ++ functionality when they were added to begin with, but it didn't happen for many reasons. That doesn't mean they should stay the way they are. Things get changed all the time because of how they impact the pack, GT++ is the same. Removing BM restorer, changing pams salt, new plastic for lategame, bee changes, hatch changes, etc... They're all new changes, but needed ones for the balance of the pack. GT++ is the same. 

Also, redoing what AE? You mean the autocrafting of the multi block controllers/casings? That's 2-3 recipes per multi, that's not even a big change. Not to mention you don't even need autocrafting for most of the controllers. That's not nearly a big enough change to counter recipe adjustments. They happen all the time, changing LuV-UV parts was something no one seemed to mind (besides the impact on mk1 fusion which was addressed), despite it being a much more substantial change as it's in the ass line which probably has more than just simple AE automation on it. 

>Why does it need to be so exponentially expensive though? You're just going to turn people away from using the whole mod because it's rediulously expensive for no real reason

I'm not sure what you mean is exponentially expensive, it's 1 core per tier, it's kind of linear. The cost of the core + multi + hatches is well within reason for what they would do given the speed change. They're still better than singleblocks, and PAs, and upgrading them vs upgrading a PA is kind of reasonable. Instead of 64x machines 3 tiers behind your power hatch, it's like 1-2 machines of your tier. It's not even that expensive. IV has multis like the ass line, which dwarf all these costs, while giving almost no use beyond simply being needed. The fusion reactors are crazy compared to these. They aren't that bad for EV-LuV multis, they're kind of cheap tbh. Especially considering that while being EV-LuV multis, they have functionality that makes them the dominant forms of whatever they are until the end of the game. Which in NH consists of 3+ tiers, ~6 for the ones in EV. That's like if the water tank you made in stone age was the best water source and was very useful until IV. Even if all of them went to LuV, that's still zpm/UV/UHV, kind of UEV, in which these are what you're using. Nothing else really scales like that, besides stuff not from GT that you WA. You can't make a nuke in EV and simply add a better energy hatch to it each tier until it's making ZPM, that's not how progression in this pack works. 

>Leave the nerfs out of the small modpack updates. They don't have to be done right now.

You're right that they have a better place, and they should come with the additions of these mods, but we don't always get it right the first time. I see absolutely no reason to wait any amount of time to change these issues if we've realised what they are and have changes proposed. 



TL;DR I don't see how this changes much in terms of redoing setups. There's some very niche circumstances but overall I don't see the problem. I don't get nerfs needing to wait, add them when the problem is noticed, otherwise you get weirdly unbalanced pockets in updates in which problems go unsolved for, uh, some reason I don't get. This isn't the first time this happened, it won't be the last. A nerf isn't different than an addition or a buff, it's simply a change. Changing between versions is the point of having a new version.

**Comment by Paolkageist** [comment]
You're too focused on getting everything to be "equal".  That doesn't exist in a modpack with 240+ mods.  There will always be some imbalance; you're the only one who sees it as a problem that has to be tackled right now.  Everyone else sees the constant nerfs and changing recipes as a problem.  And judging by multiple people in just the few comments here saying it's disruptive, I think it's a bit more than "niche circumstances" where your nerfs affect things.  It may not be disruptive to the way you play, but it obviously is to other people.  

If something's completely busted, like a new multiblock giving 10x ore output or duplicating items, then that's worth a fix; otherwise let it go and let people progress without spending every update redoing everything.  And these kinds of nerfs are definitely different than additions or buffs.  Adding a mod to the pack (like bartworks) doesn't affect what I have set up right now.  All my systems still work with the addition of bartworks.  Nerfing a multiblock by changing what it needs to even form, like adding the control core, definitely affects my current setup.  Your idea of moving AE2 to LuV or ZPM definitely affects my EV/IV setup.  These "fixes" are more gamebreaking than any imbalance that may exist in the modpack in its current form.  Hell, look at what "balancing" hatches did in 6.2 -- it's impossible to progress into LuV because the materials weren't taken into consideration.  We don't need more of that shit.  Leave it be.  Live with imbalance.  It won't kill you.

**Comment by 0lafe** [comment]
>You're too focused on getting everything to be "equal". That doesn't exist in a modpack with 240+ mods. There will always be some imbalance; you're the only one who sees it as a problem that has to be tackled right now. Everyone else sees the constant nerfs and changing recipes as a problem. And judging by multiple people in just the few comments here saying it's disruptive, I think it's a bit more than "niche circumstances" where your nerfs affect things. It may not be disruptive to the way you play, but it obviously is to other people.

You're really upset about 3-4 recipes changing in a version for a machine that was never properly implemented. You do realise this is an unfinished pack and like most things are subject to change. 

>Adding a mod to the pack (like bartworks) doesn't affect what I have set up right now

except it changes the nature of lategame circuits/plastics and forces you to setup infrastructure in a way you didn't before to make items you've already made. If anything this is a significantly larger disruption as it changes what  you can and cannot make while you stay in the same progression point, Changing gt++ multi speeds doesn't change what you have set up right now, but in a different way. It doesn't stop you from making anything you could make before, it simply adjusts the ability of the machines to more accurately fit the game. 

>All my systems still work with the addition of bartworks. Nerfing a multiblock by changing what it needs to even form, like adding the control core, definitely affects my current setup

changing multis mostly just slows you down, while adding cores forces you to bring your multis up to the current status of a completed multi. There's a reason for these changes, the multis were too cheap to upgrade before. You got away with using a fairly unbalanced multi for a while, and now you're complaining that it isn't staying forever? That's kind of ridiculous. If anything you had an advantage from using these multis in their current state, that's not something to be mad about. 

Also not any automation of wetwares/things that will use the new plastics. Those won't work, because like the multi changes, we found a better solution to the problem of lategame plastics, or circuits, and fixed it through recipe and functionality changes. This is in no way unique to gt++, and the change to the ++ multis is probably the least impactful one in terms of changing what you can and cannot make in a certain spot, pre and post change. 

>Hell, look at what "balancing" hatches did in 6.2 -- it's impossible to progress into LuV because the materials weren't taken into consideration

I'm not quite sure what you mean. Wasn't the problem with making the hatches some assline bug? If you mean there was a eu conflict with something then I'd assume it's been changed now, that's the point of a beta release. There were expected to be bugs, they were found, and then changed. 

>Leave it be. Live with imbalance. It won't kill you.

So don't change a blatant problem in the progression, because changing the requirements of a multi is the worst thing you can do to someone? I still don't understand this part really. Changing the functionality of stuff between updates happens fairly frequently, I don't see how it's this big of a deal for GT++. Also, what's so bad about forcing you to now make new hatches for your multis? If they're changed to be added to the multi cost, everyone should have to make them? Keeping them as they are simply because it's been that way doesn't really make sense in any way. The difficulty associated with crafting the hatches is completely intended if you've made the multis before or not. The change is meant to reflect their cost, so of course you have to make hatches for your multis to work now. Maybe if your entire base is running on solely gt++ multis there could be a problem? So what about this. Use recipes that only require what the energy hatches cost. This way you're able to make any core you can make an energy hatch for, and as those require singleblocks of that tier before you can upgrade your ++ multis, there's no additional singleblock reqs for the cores. This way the only additional cost that you'll run into post core update is needing to make cores that you have the machines to make, with materials you can already make. The only thing against this would be that the core cost doesn't reflect the utility of the increased overclock efficiency from parallel processing, which I personally disagree with but I guess I see the side of it being cheaper. However these multis are good, and stay good when overclocked essentially tiering up each time they get more power. That kind of needs a cost, even if it means you'll have to lose some more materials for it, that's kind of how everything works in NH already though, right? There's a fairly good reason to change the way these multis work, and it makes them fit much nicer with advancing tiers, and the themes of nh, The only additional cost to you would be creating the same item everyone else has to make, so you're not at a disadvantage or something for using the multis before the change. The point of this change is to increase the cost of the multis, and you seem to be complaining that there's an increased cost. I can understand that, but the idea of needing to keep everything the same isn't something that works for nh. Sometimes it does get a little intrusive, but this change wouldn't even be one of those times. You simply need to make a new block/item that everyone else who wants to use the same multi will need to make. As the setups that are broken aren't the ones that will be relied on to make the busses/cores to fix those setups, I really really don't see the issue here, beyond you don't want to make more items for your fairly OP machine. Something I can't really empathize with to be honest

**Comment by draknyte1** [comment]
> BlameOtherPeopleForYourProblems - Yesterday at 19:43
> Write a proposal, submit that for discussion

I didn't ask for a discussion, leading to several emails while I slept.
